src/ctia/domain/id.clj

```diff
@@ -70,13 +73,14 @@
 
 (defn short-id->id
   [type short-id {:keys [hostname path-prefix port protocol]}]
-  (map->CtiaId
-   {:hostname hostname
-    :short-id short-id
-    :path-prefix (not-empty path-prefix)
-    :port port
-    :protocol protocol
-    :type (name type)}))
+  (if (valid-short-id? short-id)
```

Comment:
I confess I normally use when in this case (there is no alternate branch).  But reality shows I probably use whatever pops out of my brain first, as I did in this case.